Build failure at Tesseract

┌ tesseract git  ...................................... [Recently updated]
├ Running autogen...
├ Running configure...
├ Running make...
Likely error (tail of the failed operation logfile):
C:/FBuild/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/FBuild/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x7e): undefined reference to `__imp_CertCloseStore'
C:/FBuild/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/FBuild/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x12f): undefined reference to `__imp_CertOpenSystemStoreW'
C:/FBuild/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/FBuild/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x147): undefined reference to `__imp_CertFreeCertificateContext'
C:/FBuild/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/FBuild/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x462): undefined reference to `__imp_CertFindCertificateInStore'
C:/FBuild/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/FBuild/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x697): undefined reference to `__imp_CertFreeCertificateContext'
C:/FBuild/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/FBuild/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x777): undefined reference to `__imp_CertFindCertificateInStore'
collect2.exe: error: ld returned 1 exit status
make[1]: *** [Makefile:4377: tesseract.exe] Error 1
make[1]: Leaving directory '/build/tesseract-git/build-64bit'
make: *** [Makefile:7985: all-recursive] Error 1
make failed. Check C:/FBuild/build/tesseract-git/build-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

Attach C:\FBuild\build\logs.zip to the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.

[logs.zip](https://github.com/user-attachments/files/26815311/logs.zip)

Same issue here. 

[logs.zip](https://github.com/user-attachments/files/26850651/logs.zip)

```
 ┌ tesseract git  ............................ [Recently updated]
 ├ Running autogen...
 ├ Running configure...
 ├ Running make...
Likely error (tail of the failed operation logfile):
C:/media-autobuild_suite/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0xdc): undefined reference to `__imp_CertFindCertificateInStore'
C:/media-autobuild_suite/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x27e): undefined reference to `__imp_CertFreeCertificateContext'
C:/media-autobuild_suite/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x673): undefined reference to `__imp_CertFreeCertificateContext'
C:/media-autobuild_suite/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x68e): undefined reference to `__imp_CertCloseStore'
C:/media-autobuild_suite/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x73f): undefined reference to `__imp_CertOpenSystemStoreW'
C:/media-autobuild_suite/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:/media-autobuild_suite/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x757): undefined reference to `__imp_CertFreeCertificateContext'
collect2.exe: error: ld returned 1 exit status
make[1]: *** [Makefile:4377: tesseract.exe] Error 1
make[1]: Leaving directory '/build/tesseract-git/build-64bit'
make: *** [Makefile:7985: all-recursive] Error 1
make failed. Check C:/media-autobuild_suite/build/tesseract-git/build-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
00:27:15   Creating diagnostics file...

Attach C:\media-autobuild_suite\build\logs.zip to the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.
```

Same issue

[logs.zip](https://github.com/user-attachments/files/26874990/logs.zip)

```bash
┌ tesseract git  ............................... [Updates found]
├ Running autogen...
├ Running uninstall...
├ Running configure...
├ Running make...
Likely error (tail of the failed operation logfile):
L:/autobuild/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: L:/autobuild/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0xdc): undefined reference to `__imp_CertFindCertificateInStore'
L:/autobuild/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: L:/autobuild/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x27e): undefined reference to `__imp_CertFreeCertificateContext'
L:/autobuild/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: L:/autobuild/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x673): undefined reference to `__imp_CertFreeCertificateContext'
L:/autobuild/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: L:/autobuild/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x68e): undefined reference to `__imp_CertCloseStore'
L:/autobuild/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: L:/autobuild/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x73f): undefined reference to `__imp_CertOpenSystemStoreW'
L:/autobuild/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: L:/autobuild/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x757): undefined reference to `__imp_CertFreeCertificateContext'
collect2.exe: error: ld returned 1 exit status
make[1]: *** [Makefile:4377: tesseract.exe] Error 1
make[1]: Leaving directory '/build/tesseract-git/build-64bit'
make: *** [Makefile:7985: all-recursive] Error 1
make failed. Check L:/autobuild/build/tesseract-git/build-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
02:26:18   Creating diagnostics file...

Attach L:\autobuild\build\logs.zip to the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.

Try running the build again at a later time.
```

Similar issue here,

└ leptonica-git  ............................................... [Updated]
├ Installing pango... done
├ Installing asciidoc... done
  Running git clone for tesseract ...
  Running git update for tesseract ...
┌ tesseract git  ...................................... [Recently updated]
├ Running autogen...
├ Running configure...
├ Running make...
Likely error (tail of the failed operation logfile):
D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0xdc): undefined reference to `__imp_CertFindCertificateInStore'
D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x27e): undefined reference to `__imp_CertFreeCertificateContext'
D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x673): undefined reference to `__imp_CertFreeCertificateContext'
D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x68e): undefined reference to `__imp_CertCloseStore'
D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x73f): undefined reference to `__imp_CertOpenSystemStoreW'
D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/MABS/msys64/ucrt64/bin/../lib/gcc/x86_64-w64-mingw32/15.2.0/../../../../lib/libcrypto.a(libdefault-lib-winstore_store.obj):(.text+0x757): undefined reference to `__imp_CertFreeCertificateContext'
collect2.exe: error: ld returned 1 exit status
make[1]: *** [Makefile:4377: tesseract.exe] Error 1
make[1]: Leaving directory '/build/tesseract-git/build-64bit'
make: *** [Makefile:7985: all-recursive] Error 1
make failed. Check D:/MABS/build/tesseract-git/build-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

Attach D:\MABS\build\logs.zip to the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.

Try running the build again at a later time.

[logs.zip](https://github.com/user-attachments/files/26894779/logs.zip)

[ab-suite.make.log](https://github.com/user-attachments/files/26894802/ab-suite.make.log)

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -740,6 +740,20 @@ if { [[ $ffmpeg != no || $standalone = y ]] && enabled libtesseract; } ||
             s| -L'"${MINGW_PREFIX}"'/lib||g
             s|'"${MINGW_PREFIX}"'/lib/lib(.+)\.dll\.a|-l\1|g
         }'
+    if nm "$MINGW_PREFIX/lib/libarchive.a" | grep __imp_zlibVersion >/dev/null 2>&1; then
+        mv -f "$MINGW_PREFIX/lib/libarchive.a" "$MINGW_PREFIX/lib/libarchive.a.bak"
+        objcopy --redefine-sym __imp_zlibVersion=zlibVersion --redefine-sym __imp_crc32=crc32 \
+            --redefine-sym __imp_inflate=inflate --redefine-sym __imp_inflateEnd=inflateEnd \
+            --redefine-sym __imp_inflateInit2_=inflateInit2_ --redefine-sym __imp_inflateReset=inflateReset \
+            --redefine-sym __imp_inflateSetDictionary=inflateSetDictionary \
+            --redefine-sym __imp_inflateInit_=inflateInit_ \
+            --redefine-sym __imp_deflate=deflate --redefine-sym __imp_deflateEnd=deflateEnd \
+            --redefine-sym __imp_deflateInit2_=deflateInit2_ --redefine-sym __imp_deflateReset=deflateReset \
+            --redefine-sym __imp_deflateSetDictionary=deflateSetDictionary \
+            --redefine-sym __imp_deflateInit_=deflateInit_ \
+            "$MINGW_PREFIX/lib/libarchive.a.bak" "$MINGW_PREFIX/lib/libarchive.a"
+        rm -f "$MINGW_PREFIX/lib/libarchive.a.bak"
+    fi
 fi
 
 if [[ $ffmpeg != no || $standalone = y ]] && enabled libtesseract; then
@@ -765,6 +779,7 @@ if [[ $ffmpeg != no || $standalone = y ]] && enabled libtesseract; then
         do_uninstall include/tesseract "${_check[@]}"
         sed -i 's|Requires.private.*|& libarchive iconv libtiff-4|' tesseract.pc.in
         grep_or_sed ws2_32 "$MINGW_PREFIX/lib/pkgconfig/libarchive.pc" 's;Libs.private:.*;& -lws2_32;g'
+        grep_or_sed crypt32 "$MINGW_PREFIX/lib/pkgconfig/libarchive.pc" 's;Libs.private:.*;& -lcrypt32;g'
         case $CC in
         *clang) sed -i -e 's|Libs.private.*|& -fopenmp=libomp|' tesseract.pc.in ;;
         *) sed -i -e 's|Libs.private.*|& -fopenmp -lgomp|' tesseract.pc.in ;;

